gc.c

```diff
@@ -4673,7 +4673,7 @@ id2ref(VALUE objid)
         }
     }
 
-    if ((orig = id2ref_obj_tbl(objspace, objid)) != Qundef &&
+    if (!UNDEF_P((orig = id2ref_obj_tbl(objspace, objid))) &&
```

Comment:
There are 12 excess parentheses in 8 files.
```suggestion
    if (!UNDEF_P(orig = id2ref_obj_tbl(objspace, objid)) &&
```